tool/travis.sh

```diff
@@ -0,0 +1,84 @@
+#!/bin/bash
+
+# Fast fail the script on failures.
+set -e
+
+# Check arguments.
+TASK=$1
+
+if [ -z "$TASK" ]; then
+  echo -e '\033[31mTASK argument must be set!\033[0m'
+  echo -e '\033[31mExample: tool/travis.sh test:unit\033[0m'
+  exit 1
+fi
+
+DART_VERSION=$(dart --version 2>&1)
+DART_2_PREFIX="Dart VM version: 2"
+
+# Run the correct task type.
+case $TASK in
+  test:unit)
+    echo -e '\033[1mTASK: Testing [test]\033[22m'
+
+    echo -e 'pub build test --mode=debug --web-compiler=dartdevc'
+    # Precompile tests to avoid timeouts/hung builds.
+    pub build test --mode=debug --web-compiler=dartdevc
```

Comment:
This might make a good example to follow for other test suites that are timing out.

Reply:
Yeah I was going to try this out next in one of our lower-level internal packages to make sure it works on smithy, and I think Greg was going to try it out in WSD as well.